doc/source/fq_default.rst

```diff
@@ -104,10 +104,13 @@ Context Management
 
     Prints the context information to ``stdout``.
 
-.. function:: void fq_default_ctx_randtest(fq_default_ctx_t ctx)
+.. function:: void fq_default_ctx_init_randtest(fq_default_ctx_t ctx, flint_rand_t state, int type)
 
-    Initializes ``ctx`` to a random finite field.  Assumes that
-    ``fq_default_ctx_init`` has not been called on ``ctx`` already.
+    Initializes ``ctx`` to a random finite field where the prime and degree is
+    set according to ``type``.
+    Assumes that ``fq_default_ctx_init`` has not been called on ``ctx`` already.
+    To see what prime and degrees may be output, see
+    ``type`` in :func:`_nmod_poly_conway_rand`.
```

Comment:
Just to be clear: You are not referring to the internal "type" for `fq_default` (i.e. `fmpz`, `nmod` etc.)?

I have to think about the randomness thing (i.e. which type). Now, this module is not very much tested inside FLINT, unfortunately (however it is tested in Nemo/Oscar), but in general we would like to (1) minimize the time for testing and (2) reduce the complexity of usage. A separate parameter for `type` obviously increases the complexity, and, furthermore, is inherited from `nmod_poly`.

Reply:
The type here is the one passed into the generation of prime,degree from the nmod conway function and is not related at all the the fq_default type

The design here, hopefully, simply does what practically would happen if you asked for a random context from fq, fq_nmod and fq_zech